mindsdb/api/executor/sql_query/steps/join_step.py

```diff
@@ -74,7 +74,11 @@ def adapt_condition(node, **kwargs):
                     return Identifier(parts=['table_b', col_name])
 
             if step.query.condition is None:
-                raise NotSupportedYet('Unable to join table without condition')
+                # prevent memory overflow
+                if len(left_data) * len(right_data) < 10 ** 7:
```

Comment:
Are `left_data` and `right_data` dataframes? If so, then may be better to get real size (`df.memory_usage(index=True, deep=True).sum()`) and compare with free memory?

Reply:
They are [ResultSets](https://github.com/mindsdb/mindsdb/blob/1f4907b9041b3f5ebc8bc2f6ba12598553588371/mindsdb/api/executor/sql_query/result_set.py#L38)